crates/anvil/src/eth/miner.rs

```diff
@@ -67,6 +75,12 @@ impl Miner {
         cx: &mut Context<'_>,
     ) -> Poll<Vec<Arc<PoolTransaction>>> {
         self.inner.register(cx);
+        // Add replay transactions to the pool
+        if let Some(replay_transactions) = self.replay_transactions.take() {
+            replay_transactions.into_iter().for_each(|tx| {
+                pool.add_transaction(tx).unwrap();
+            });
+        }
         self.mode.write().poll(pool, cx)
```

Comment:
we need to change this logic slightly, something like

```rust
let next = ready!(self.mode.write().poll(pool, cx))
// check if we have replay tx, if so return [replay.., next..]
// this way we're able to mine a block with the replay tx and txs in the pool
```

Reply:
updated